tower/src/limit/concurrency/service.rs

```diff
@@ -26,9 +26,14 @@ pub struct ConcurrencyLimit<T> {
 impl<T> ConcurrencyLimit<T> {
     /// Create a new concurrency limiter.
     pub fn new(inner: T, max: usize) -> Self {
+        Self::new_shared(inner, Arc::new(Semaphore::new(max)))
+    }
+
+    /// Create a new concurrency limiter with a provided shared semaphore
+    pub fn new_shared(inner: T, semaphore: Arc<Semaphore>) -> Self {
```

Comment:
nitpicky, TIOLI: if we're going to have this API, i might prefer naming it something like:
```suggestion
    pub fn with_semaphore(inner: T, semaphore: Arc<Semaphore>) -> Self {
```

Reply:
on one hand, like my suggestion above, not having `tokio::sync`'s semaphore in the public API might be theoretically nicer, hiding the implementation details. on the other hand, this API is the only way we can share a semaphore across`ConcurrencyLimit` services _without_ using the `Layer`, so maybe having this constructor is worth having the semaphore exposed publicly.

if we decide to keep this public, and i think maybe we should, it's fine for the `GlobalConcurrencyLimitLayer` constructor to take a semaphore as it does currently...